src-frontend/app/controllers/procedure.js

```diff
@@ -63,6 +64,29 @@ export default Ember.ObjectController.extend({
             });
 
             this.endPropertyChanges();
+        },
+        generateProtocol: function() {
+            var filename = this.get('model.title') + '.xml';
+            $.ajax({
+                beforeSend: function(xhr) {
+                    xhr.setRequestHeader('Authorization', 'Token ' + Cookies.get('authorizationToken'));
+                },
+                type: 'GET',
+                url: ENV.APP.API_HOST + '/' + ENV.APP.API_NAMESPACE + '/procedures/' + this.get('model.id') + '/generate/',
+                success: function(data) {
+                    var blob = new Blob([(new XMLSerializer()).serializeToString(data)], { type: 'text/xml' });
+                    var url = window.URL.createObjectURL(blob);
+                    var a = document.createElement("a");
+                    var randomId = Math.random().toString();
```

Comment:
Why do we need to use a random number string for an ID vs simply giving it an explicit ID?